sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/SqlTraversalEngine.java

```diff
@@ -100,9 +99,20 @@ public static void addProjection(final List<GremlinSqlIdentifier> gremlinSqlIden
     }
 
 
-    public static void applyTraversal(final GraphTraversal<?, ?> graphTraversal,
-                                      final GraphTraversal<?, ?> subGraphTraversal) {
-        graphTraversal.by(subGraphTraversal);
+    public static void applyTraversal(boolean apply,
+                                      final GraphTraversal graphTraversal,
+                                      final GraphTraversal subGraphTraversal) {
+        if (apply) {
+            graphTraversal.by(__.coalesce(subGraphTraversal, __.constant(SqlGremlinQueryResult.NULL_VALUE)));
+        } else {
+            graphTraversal.by(subGraphTraversal);
+        }
+    }
+
+
+    public static void applyTraversal(final GraphTraversal graphTraversal,
+                                      final GraphTraversal subGraphTraversal) {
+        applyTraversal(false, graphTraversal, subGraphTraversal);
```

Comment:
Nit: Standard Java practices usually have overloaded methods with the missing parameter at the end. `apply` should be last here.